src/Courses/Settings/OverridesForm.tsx

```diff
@@ -0,0 +1,238 @@
+            const res = await extendTopic({courseTopicContentId, userId, extensions});
+            console.log(res);
+        } catch (e) {
+            console.error(e);
+            throw e;
+        }
+    };
+
+    const updateQuestions = async (courseTopicQuestionId: number, userId: number, extensions: {maxAttempts: number}) => {
+        try {
+            const res = await extendQuestion({courseTopicQuestionId, userId, extensions});
+            console.log(res);
+        } catch (e) {
+            console.error(e);
+            throw e;
+        }
+    };
+
+    const onSubmit = async (extensions: {startDate: Moment, endDate: Moment, deadDate: Moment} | {maxAttempts: number}) => {
+        setSubmitLoading(true);
+        console.log(extensions);
+        if (problem) {
+            await updateQuestions(problem.id, userId, extensions as {maxAttempts: number});
+        } else if (topic) {
+            await updateTopic(topic.id, userId, extensions as {startDate: Moment, endDate: Moment, deadDate: Moment});
+        } else {
+            console.error('Unhandled override case.');
+        }
+        setSubmitLoading(false);
+    };
+
+    const renderQuestionOverrideForm = (question: ProblemObject) => (
+        <Grid item container md={12} alignItems='flex-start' justify="center">
+            <Grid item md={4}>
+                <TextField inputRef={register} name="maxAttempts" defaultValue={question.maxAttempts} label='Max Attempts' />
+            </Grid>
+        </Grid>
+    );
+
+    const renderTopicOverrideForm = (topic: NewCourseTopicObj) => (
+        <>
+            <Grid item md={12}>
+                <MuiPickersUtilsProvider utils={MomentUtils}>
+                    <Controller
+                        as={<KeyboardDatePicker value="" onChange={() => {}} />}
+                        name="startDate"
+                        control={control}
+                        defaultValue={moment(topic.startDate)}
+                        autoOk
+                        variant="inline"
+                        format="MM/DD/yyyy"
+                        KeyboardButtonProps={{
+                            'aria-label': 'change date',
+                        }}
+                        fullWidth={true}
+                        label='Start Date'
+                        InputLabelProps={{style: { color: 'rgba(255, 255, 255, 0.8)', fontSize: drawerFontSize }}}
+                        inputProps={{ style: { textAlign: 'center', color: 'rgba(255, 255, 255, 0.8)', fontSize: drawerFontSize } }}
+                    />
+                </MuiPickersUtilsProvider>
+            </Grid>
+
+            <Grid item md={12}>
+                <MuiPickersUtilsProvider utils={MomentUtils}>
+                    <Controller
+                        as={<KeyboardDatePicker value="" onChange={() => {}} />}
+                        name="endDate"
+                        control={control}
+                        defaultValue={moment(topic.endDate)}
```

Comment:
This should be value and not default value, otherwise when updates come into props it won't update (though I think the controller is a problem anyway)